gapis/api/vulkan/vulkan.go

```diff
@@ -198,10 +199,19 @@ func (API) GetDependencyGraphBehaviourProvider(ctx context.Context) dependencygr
 	return newVulkanDependencyGraphBehaviourProvider()
 }
 
-func (API) MutateSubcommands(ctx context.Context, id api.CmdID, cmd api.Cmd, s *api.State, callback func(*api.State, api.SubCmdIdx, api.Cmd)) error {
+func (API) MutateSubcommands(ctx context.Context, id api.CmdID, cmd api.Cmd,
+	s *api.State, preSubCmdCb func(*api.State, api.SubCmdIdx, api.Cmd),
+	postSubCmdCb func(*api.State, api.SubCmdIdx, api.Cmd)) error {
 	c := GetState(s)
-	c.HandleSubcommand = func(_ interface{}) {
-		callback(s, append(api.SubCmdIdx{uint64(id)}, c.SubCmdIdx...), cmd)
+	if postSubCmdCb != nil {
+		c.PostSubcommand = func(_ interface{}) {
```

Comment:
nit: you can drop the `_` from the parameter name if there's no other named parameters:
```go
c.PostSubcommand = func(interface{}) {
```